libs/hwes/event.go

```diff
@@ -2,29 +2,124 @@ package hwes
 
 import (
 	"encoding/json"
+	"fmt"
 	"github.com/EventStore/EventStore-Client-Go/esdb"
 	"github.com/google/uuid"
+	"strings"
 	"time"
 )
 
 type Event struct {
-	ID        uuid.UUID
-	Type      string
-	Data      []byte
-	Timestamp time.Time
+	EventID       uuid.UUID
+	EventType     string
+	AggregateID   uuid.UUID
+	AggregateType AggregateType
+	Data          []byte
+	Timestamp     time.Time
+	Version       uint64
 }
 
-func NewBaseEvent(eventType string) Event {
+func NewEventBase(aggregate Aggregate, eventType string) Event {
 	return Event{
-		ID:        uuid.New(),
-		Type:      eventType,
-		Timestamp: time.Now().UTC(),
+		EventID:       uuid.New(),
+		EventType:     eventType,
+		AggregateID:   aggregate.GetID(),
+		AggregateType: aggregate.GetType(),
+		Timestamp:     time.Now().UTC(),
 	}
 }
 
+func NewEventBaseWithPayload(aggregate Aggregate, eventType string, payload interface{}) (Event, error) {
```

Comment:
mention that the payload will be json marshalled. that's important information for a user of this function

maybe just mention `SetJsonData`